pkg/provisioner/ironic/ironic.go

```diff
@@ -107,37 +148,70 @@ func LogStartup() {
 // A private function to construct an ironicProvisioner (rather than a
 // Provisioner interface) in a consistent way for tests.
 func newProvisioner(host *metal3v1alpha1.BareMetalHost, bmcCreds bmc.Credentials, publisher provisioner.EventPublisher) (*ironicProvisioner, error) {
-	client, err := noauth.NewBareMetalNoAuth(noauth.EndpointOpts{
-		IronicEndpoint: ironicEndpoint,
-	})
-	if err != nil {
-		return nil, err
+	var provisioner ironicProvisioner
+
+	if authStrategy == "http_basic" {
```

Comment:
There are some things repeated in this function, too. For example, setting the required micro version of the client should only be done in one place, so we don't miss one if we have to update the value.

If the only difference between the clients is the type used, could we construct the clients in the conditional and then configure them outside the conditional before building the provisioner we're going to return?

Reply:
Will look on how to improve this, thanks!